include/boost/math/tools/simple_continued_fraction.hpp

```diff
@@ -32,34 +36,59 @@ namespace boost::math::tools {
 template<typename Real, typename Z = int64_t>
 class simple_continued_fraction {
 public:
-    simple_continued_fraction(Real x) : x_{x} {
+    typedef Z int_type;
+
+    simple_continued_fraction(std::vector<Z> data) : b_{std::move(data)} {
```

Comment:
```suggestion
    simple_continued_fraction(std::vector<Z>&& data) : b_{data} {
```

As written this makes a copy of the data vector and then moves it into `b_`. I expect you want to take an r-value reference instead.

Reply:
`simple_continued_fraction(std::vector<Z>)` can be used both with `std::vector<Z>&&` and `const std::vector<Z>&` arguments, whereas `simple_continued_fraction(std::vector<Z>&&)` cannot be used with `const std::vector<Z>&`. In the case of `std::vector<Z>&&` argument, there should no significant difference (`std::vector<Z>` can require one more call to move constructor, which is negligible).

More importantly, I'm pretty sure that your suggestion `simple_continued_fraction(std::vector<Z>&& data) : b_{data}` will call copy constructor on `data`, because you pass it as an lvalue, not as an rvalue.